frontend/src/metabase/browse/containers/BrowseModels.tsx

```diff
@@ -0,0 +1,278 @@
+  getHowLongAgo,
+  type ItemWithLastEditInfo,
+} from "metabase/components/LastEditInfoLabel/LastEditInfoLabel";
+import type { Card, Collection, SearchResult } from "metabase-types/api";
+import * as Urls from "metabase/lib/urls";
+
+import Link from "metabase/core/components/Link";
+import LastEditInfoLabel from "metabase/components/LastEditInfoLabel";
+import LoadingAndErrorWrapper from "metabase/components/LoadingAndErrorWrapper";
+
+import type { useSearchListQuery } from "metabase/common/hooks";
+
+import { ANALYTICS_CONTEXT } from "metabase/browse/constants";
+
+import NoResults from "assets/img/no_results.svg";
+import { Icon, Text } from "metabase/ui";
+import { formatDateTimeWithUnit } from "metabase/lib/formatting";
+import {
+  CenteredEmptyState,
+  CollectionHeaderContainer,
+  GridContainer,
+  LastEditedInfoSeparator,
+  ModelCard,
+  MultilineEllipsified,
+} from "./BrowseData.styled";
+
+const emptyArray: SearchResult[] = [];
+
+export const BrowseModels = ({
+  data: models = emptyArray,
+  error,
+  isLoading,
+}: ReturnType<typeof useSearchListQuery<SearchResult>>) => {
+  useEffect(() => {
+    const configureGrid = () => {
+      const gridOptions = getGridOptions(models);
+      setGridOptions(gridOptions);
+    };
+    configureGrid();
+    window.addEventListener("resize", configureGrid);
+    return () => window.removeEventListener("resize", configureGrid);
+  }, [models]);
+
+  const [gridOptions, setGridOptions] = useState<{
+    cells: Cell[];
+  } | null>(null);
+
+  if (error) {
+    return <LoadingAndErrorWrapper error />;
+  } else if (isLoading || !gridOptions) {
+    return (
+      <LoadingAndErrorWrapper loading style={{ display: "flex", flex: 1 }} />
+    );
+  }
+
+  const { cells = [] } = gridOptions;
+
+  return cells.length ? (
+    <GridContainer role="grid">{cells}</GridContainer>
+  ) : (
+    <CenteredEmptyState
+      title={t`No models here yet`}
+      message={t`Models help curate data to make it easier to find answers to questions all in one place.`}
+      illustrationElement={<img src={NoResults} />}
+    />
+  );
+};
+
+interface ModelCellProps {
+  model: SearchResult;
+  style?: React.CSSProperties;
+  collectionHtmlId: string;
+}
+
+const ModelCell = ({ model, style, collectionHtmlId }: ModelCellProps) => {
+  const modelWithHistory = addLastEditInfo(model);
+  const lastEdit = modelWithHistory["last-edit-info"];
+  const lastEditorName = lastEdit.full_name;
+  const howLongAgo = getHowLongAgo(lastEdit.timestamp);
+
+  const headingId = `heading-for-model-${model.id}`;
+
+  const formattedDate = formatDateTimeWithUnit(lastEdit.timestamp, "day", {});
+
+  const time = <time dateTime={lastEdit.timestamp}>{formattedDate}</time>;
+  const tooltipLabel = jt`Last edited by ${lastEditorName}${(<br />)}${time}`;
+
+  return (
+    <Link
+      aria-labelledby={`${collectionHtmlId} ${headingId}`}
+      key={model.id}
+      style={style}
+      to={Urls.model(model as unknown as Partial<Card>)}
```

Comment:
Please correct me if I'm wrong, but I believe some fudging with the types like this is needed, because a model is represented as a `SearchResult` or `CollectionItem` by some portions of the codebase (since models are retrieved via `/api/search`), and as a `Card` in other portions.